### ISSUE TITLE
Firefox goes unresponsive when trying to import feeds

### ISSUE THREAD

**Issue by fritzophrenic** [nt1m/livemarks#87]
When I installed Firefox 64, which dropped native feed support, it automatically exported a OPML file to my desktop containing all my old feed subscriptions. I wanted to import this file into Livemarks to make my transition to an extension easy and fast.

I went to the Add-ons Manager in Firefox and clicked "options" button next to Livemarks extension. From there, I click "More Options" where there is an Import/export section. Clicking the "Import feeds" button pops up what I assume will be a file browser. However, the popup window never populates, instead, it goes unresponsive, and brings Firefox down with it. I need to force-kill Firefox.

The same behavior happens after manually adding one of the feeds from my OPML export file, using the "+ New livemark" button instead of "More Options".

In case it matters, I changed the Default folder to the location all my old feeds were stored, and I changed the refresh interval to 120 minutes. I made no other setting changes.

I tried this multiple times over about a half hour, with the same behavior each time. I tried opening the livemarks options tab in a new window by itself, as I have dozens of tabs open already (and I'm not keen on abandoning all of them right now).

**Comment by nt1m** [comment]
Can you upload your opml file ?

**Comment by fritzophrenic** [comment]
I removed one entry that contained my Github username and some sort of token (which wasn't working anyway) but otherwise, here it is.

Note I never got to the point where I was able to select this file in the import dialog. The import dialog UI never finished loading.

[Firefox feeds backup.redacted.opml.txt](https://github.com/nt1m/livemarks/files/2672812/Firefox.feeds.backup.redacted.opml.txt)

**Comment by nt1m** [comment]
Thanks! Do you also have a crash report from about:crashreports ?

**Comment by fritzophrenic** [comment]
I didn't know about about:crashes! Sadly I guess Firefox didn't "crash" hard enough to count, I don't see any data for today on that page. I got the "Firefox is not responding" dialog from Windows and chose to close the program. A couple times I additionally had to use task manager to completely kill the process.

Is there something I can do ahead of time to force a report before reproducing again?

I'll also try later on my home computer in case it's hardware related or depends on my open tabs or something.

**Comment by fritzophrenic** [comment]
@mdcdr, I think that's a different issue. I couldn't even get the file browser to pop up to *start* the import.

**Comment by mdcdr** [comment]
Probably you're right I'll post in #85.

**Comment by evilpie** [comment]
I think one basic problem is that for every single livemark that is imported we download the feed for all previously imported or existing feeds.

```
updateAllLivemarks Array [ "_5MvXz3tGln7" ]

updateAllLivemarks@moz-extension://03c1495b-1006-4a8a-b159-7b1991c3fc81/background/background.js:31:53
_emit@moz-extension://03c1495b-1006-4a8a-b159-7b1991c3fc81/shared/livemark-store.js:89:9
setupSync/<@moz-extension://03c1495b-1006-4a8a-b159-7b1991c3fc81/shared/livemark-store.js:63:9
```

**Comment by fritzophrenic** [comment]
I wanted to update that on the same computer I originally saw this issue on, I was able to successfully bring up the import dialog today. It looks like you might be using this issue to track generic performance issues with the import now, but my original issue seems fixed somehow. I.e., when I click "options", "more options", "import feeds", Firefox does not hang anymore. I see a file browser now, whereas before the file browser never fully appeared. Once the file browser appeared, I did not try importing, as my feeds are already in place.

As far as I'm concerned this issue can be closed.
